internal/driver/driver.go

```diff
@@ -375,9 +318,13 @@ func (d *Driver) handleWriteCommands(devName string, p protocolMap, reqs []dsMod
 		if err != nil {
 			return 0, err
 		}
-		var u uint64
-		u, err = strconv.ParseUint(v, 10, 64)
-		return u, errors.Wrapf(err, "unable to parse attribute %q with val %q as uint", key, v)
+
+		u, ok := v.(uint64)
+		if !ok {
+			return 0, fmt.Errorf("unable to cast attribute %q with val %q as uint", key, v)
+		}
+
+		return u, nil
```

Comment:
This is broken. The value returned is `float64` and is unable to be converted into a `uint64` here.

A beefier solution could be this:
```go
		switch val := v.(type) {
		case uint64:
			return val, nil
		case float64:
			return uint64(val), nil
		case string:
			u, err := strconv.ParseUint(val, 10, 64)
			if err != nil {
				return 0, fmt.Errorf("unable to convert attribute %q with val %q as uint", key, v)
			}
			return u, nil
		default:
			return 0, fmt.Errorf("unable to cast attribute %q with val %q as uint", key, v)
		}
```

However it might make more sense to re-visit why you had to change this code in the first place? I think it also might have to do with the values not being quoted in the Impinj profile yaml.

Reply:
> EDIT: I see now that it was likely changed due to reqs[idx].Attributes returning map[string]interface{} instead of map[string]string

restored values in YAML to strings and processed them similar to previous